src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt

```diff
@@ -173,7 +177,28 @@ class VisitSchedulerClient(
       .body(BodyInserters.fromValue(requestDto))
       .accept(MediaType.APPLICATION_JSON)
       .retrieve()
-      .bodyToMono<VisitDto>().block(apiTimeout)
+      .bodyToMono<VisitDto>().onErrorResume {
+          e ->
+        if (isUnprocessableEntityError(e)) {
+          val exception = getBookVisitApplicationValidationErrorResponse(e)
+          Mono.error(exception)
+        } else {
+          Mono.error(e)
+        }
+      }.block(apiTimeout)
+  }
+
+  private fun getBookVisitApplicationValidationErrorResponse(e: Throwable): Throwable {
+    if (e is WebClientResponseException && isUnprocessableEntityError(e)) {
+      try {
+        val errorResponse = objectMapper.readValue(e.responseBodyAsString, ApplicationValidationErrorResponse::class.java)
+        return ApplicationValidationException(errorResponse.validationErrors)
+      } catch (jsonProcessingException: Exception) {
+        LOG.error("An error occurred processing the application validation error response - ${e.stackTraceToString()}")
```

Comment:
weirdly i think this needs an exception haha?

Reply:
updated to throw an Exception instead.

Reply:
sounds good!